packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart

```diff
@@ -162,20 +161,21 @@ class LegacyCredentialProviderIOS implements LegacyCredentialProvider {
   }
 
   @override
-  Future<LegacyDeviceDetails?> fetchLegacyDeviceSecrets({
-    required String username,
-    CognitoUserPoolConfig? userPoolConfig,
-  }) async {
-    if (userPoolConfig == null) return null;
+  Future<LegacyDeviceDetails?> fetchLegacyDeviceSecrets(
+    String username,
+    AuthOutputs authOutputs,
+  ) async {
+    if (authOutputs.userPoolClientId == null) return null;
```

Comment:
*suggestion:* Remove the force casts. If you create variables for userPoolId and userPoolClientId and return early if they are null, then Dart will not require the force case.

```dart
    final userPoolId = authOutputs.userPoolId;
    final userPoolClientId = authOutputs.userPoolClientId;
    if (userPoolId == null || userPoolClientId == null) return null;
    final userPoolStorage = await _getUserPoolStorage();
    final cognitoUserKeys = LegacyCognitoUserKeys(userPoolClientId);
    final currentUserId = await userPoolStorage.read(
      key: cognitoUserKeys[LegacyCognitoKey.currentUser],
    );
    if (currentUserId == null) return null;
    final keys = LegacyDeviceSecretKeys(currentUserId, userPoolId);
```

Reply:
This applies to several other locations in the PR.